### PR TITLE
Rsx: Fix fog (Fixes #1535)

### LINKED ISSUE #1535: After Burner Climax, DX12, broken graphics
AMD R9 290:
image
image
Intel HD4600:
image

Started happening around https://github.com/RPCS3/rpcs3/commit/422f5c00083a40bee0aeddc18e551f2a35c4d4d5 . Haven't pinpointed exact commit yet. Affects both intel and amd.

### PR DESCRIPTION
**Pull request by vlj**
This PR fixes fog in After Burner Climax by setting linear mode as default ; it also split fog distance and fog fraction in the fog input declaration.

### CODE REVIEW

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -768,7 +768,10 @@ namespace rsx
 
 		method_registers[NV4097_SET_LINE_WIDTH] = 1 << 3;
 
-		method_registers[NV4097_SET_FOG_MODE] = 0x0800; // rsx::fog_mode::exponential;
+		// These defaults were found using After Burner Climax (which never set fog mode despite using fog input)
+		method_registers[NV4097_SET_FOG_MODE] = 0x2601; // rsx::fog_mode::linear;
+		(f32&)method_registers[NV4097_SET_FOG_MODE] = 1.;
```
Comment (danilaml) [inline rpcs3/Emu/RSX/RSXThread.cpp]: What is happening here?

Comment (vlj) [inline rpcs3/Emu/RSX/RSXThread.cpp]: After Burner never touch the NV4097_SET_FOG_MODE/NV4097_SET_FOG_PARAMS.
However it uses the fogc input in some fragment shader and according to some youtube video there is indeed fog in the long distance.

Linear/1./1. values leaves the fogc input unmodified (ie fogc = 1. * In.fogc + 1. - 1.) and the visual matches the video youtube.

BTW I fixed the typo.

Comment (danilaml) [inline rpcs3/Emu/RSX/RSXThread.cpp]: Yeah, I was just wondering why same register was assigned different values. Now it's fixed.

Comment (tambry) [inline rpcs3/Emu/RSX/RSXThread.cpp]: Does this fix the [glitchy fog](http://puu.sh/nuEgZ/1633e7032c.png) at the end of the every level/stage that would be seen?

Comment (vlj) [inline rpcs3/Emu/RSX/RSXThread.cpp]: It was already fixed in a previous PR. It was related to alpha test reference being wrongly interpreted.

Comment (danilaml) [inline rpcs3/Emu/RSX/RSXThread.cpp]: @tambry I think this should fix https://github.com/RPCS3/rpcs3/issues/1535

Comment (tambry) [inline rpcs3/Emu/RSX/RSXThread.cpp]: Added the note for fix to the title. Should automagically close the issue, if merged.

**Pull request merged**